pkg/image/api/types.go

```diff
@@ -60,6 +77,41 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob. Provided by caller.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side validation.
+	ValidatedOn unversioned.Time
+	// Result of validation performed at ValidatedOn.
+	ValidationResult SignatureValidationResult
+
+	// At most one of the following pointers can point to details. If the signature type isn't recognized,
+	// they will all be nil.
+
+	// Details for signature of type "atomic".
+	AtomicSignatureDetails *AtomicSignatureDetails
```

Comment:
> So using Jessica and the console as a stand-in, here's what I would expect: 
> ...

I very much agree.

> 1 Signer / signatory - the entity the user should be expected to trust

Not available in the "atomic" signature. `Creator` field, denoting signing library - thus having little value to the consumer, has been removed.

> 2 The item being signed - the image digest aka ImageName
> 3 A source URL / value that loosely identifies the signing value - ImageLocation or some similar value (not DockerImageReference if at all possible, esp. if it does not follow DIR rules)

`SignedDockerImageReference` has just been renamed to `SignedImageOriginName`.

> 4 The expiry, if specified

Current signature cannot expire. We can add this field later if we add support for expiring signature kind.

> 5 The signing date, if specified.